bundles/org.openhab.binding.emby/src/main/java/org/openhab/binding/emby/internal/protocol/EmbyClientSocket.java

```diff
@@ -0,0 +1,249 @@
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * EmbyClientSocket implements the low level communication to Emby through
+ * websocket. Usually this communication is done through port 9090
+ *
+ * @author Zachary Christiansen - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class EmbyClientSocket {
+
+    private final Logger logger = LoggerFactory.getLogger(EmbyClientSocket.class);
+
+    private final ScheduledExecutorService scheduler;
+    private static final int REQUEST_TIMEOUT_MS = 60000;
+
+    private @Nullable CountDownLatch commandLatch = null;
+    private @Nullable JsonObject commandResponse = null;
+
+    private boolean connected = false;
+
+    private final JsonParser parser = new JsonParser();
+    private final Gson mapper = new Gson();
+    private @Nullable URI uri;
+    private @Nullable Session session;
+    private WebSocketClient client;
+    private int bufferSize;
+
+    private final EmbyClientSocketEventListener eventHandler;
+
+    public EmbyClientSocket(EmbyClientSocketEventListener setEventHandler, @Nullable URI setUri,
+            ScheduledExecutorService setScheduler, int setBufferSize) {
+        eventHandler = setEventHandler;
+        uri = setUri;
+        client = new WebSocketClient();
+        scheduler = setScheduler;
+        bufferSize = setBufferSize;
+    }
+
+    /**
+     * Attempts to create a connection to the Emby host and begin listening for
+     * updates over the async http web socket
+     *
+     * @throws Exception
+     */
+    public synchronized void open() throws Exception {
+        if (isConnected()) {
+            logger.warn("connect: connection is already open");
+        }
+        if (!client.isStarted()) {
+            client.start();
+            client.setMaxTextMessageBufferSize(bufferSize);
+        }
+        EmbyWebSocketListener socket = new EmbyWebSocketListener();
+        ClientUpgradeRequest request = new ClientUpgradeRequest();
+
+        client.connect(socket, uri, request);
+    }
+
+    /***
+     * Close this connection to the Emby instance
+     */
+    public void close() {
+        // if there is an old web socket then clean up and destroy
+        if (session != null) {
+            try {
+                session.close();
+            } catch (Exception e) {
+                logger.debug("Exception during closing the websocket: {}", e.getMessage(), e);
+            }
+            session = null;
+        }
+        try {
+            client.stop();
+        } catch (Exception e) {
+            logger.debug("Exception during closing the websocket: {}", e.getMessage(), e);
+        }
+    }
+
+    public boolean isConnected() {
+        if (session == null || !session.isOpen()) {
+            return false;
+        }
+
+        return connected;
+    }
+
+    @WebSocket
+    @NonNullByDefault
```

Comment:
Inner classes don't need to be annotated. There is a bug in checkstyle, which generates false positives in these cases.